File: frontend/src/types/booking.ts
export type SeatState = "AVAILABLE" | "LOCKED" | "BOOKED" | "BLOCKED" | "MINE";

export type SeatCell = {
  id: number;
  rowLabel: string;
  seatNumber: string;
  label: string;
  categoryId: number;
  categoryName: string;
  price: number;
  currency: string;
  state: SeatState;
};

export type SeatMap = {
  eventId: number;
  eventTitle: string;
  eventSlug: string;
  lockTtlSeconds: number;
  seats: SeatCell[];
};

export type Booking = {
  id: number;
  bookingCode: string;
  status: "PENDING" | "CONFIRMED" | "CANCELLED" | "EXPIRED" | "REFUNDED";
  eventId: number;
  eventTitle: string;
  eventSlug: string;
  startsAt: string;
  subtotal: number;
  discountAmount: number;
  totalAmount: number;
  currency: string;
  expiresAt?: string | null;
  createdAt?: string | null;
  seats: Array<{
    id: number;
    label: string;
    categoryName?: string | null;
    price?: number | null;
  }>;
  tickets: Array<{
    id: number;
    ticketCode: string;
    qrPayload: string;
    status: string;
    seatLabel?: string | null;
  }>;
};

File: backend/src/main/java/com/aetherpass/entity/Booking.java
package com.aetherpass.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@Table(name = "bookings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "booking_code", nullable = false, unique = true, length = 40)
    private String bookingCode;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "event_id", nullable = false)
    private Event event;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "coupon_id")
    private Coupon coupon;

    @Column(nullable = false, length = 30)
    private String status; // PENDING | CONFIRMED | CANCELLED | EXPIRED | REFUNDED

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal subtotal;

    @Column(name = "discount_amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal discountAmount;

    @Column(name = "total_amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal totalAmount;

    @Column(nullable = false, length = 3)
    private String currency;

    @Column(name = "expires_at")
    private Instant expiresAt;

    @Column(name = "created_at", nullable = false, insertable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false, insertable = false, updatable = false)
    private Instant updatedAt;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "booking_seats",
            joinColumns = @JoinColumn(name = "booking_id"),
            inverseJoinColumns = @JoinColumn(name = "seat_id")
    )
    @Builder.Default
    private Set<Seat> seats = new HashSet<>();

    @OneToMany(mappedBy = "booking", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Ticket> tickets = new ArrayList<>();
}
